### PR TITLE
ECER-896: Add missing UserProfile fields to BE

### PR DESCRIPTION
**Pull request by pcatkins**
## Description

- Added missing fields `AlternateContactPhone`, `PreferredName`, and `MiddleName` to UserProfile
- Removed Previous Name field from form
- Styling changes from demo prep feedback

## Related Jira Issue(s)

- [ECER-620](https://eccbc.atlassian.net/browse/ECER-620)

## Checklist
- [X] I have tested these changes locally.
- [X] Changes to backend endpoints are covered by passing integration tests.
- [X] I have added or updated the necessary documentation.

### CODE REVIEW

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/UserInfoEndpoints.cs [hunk 1]
```diff
@@ -37,4 +37,4 @@ public void Register(IEndpointRouteBuilder endpointRouteBuilder)
   }
 }
 
-public record UserInfo(string FirstName, string LastName, DateOnly DateOfBirth, string Email, string Phone);
+public record UserInfo(string FirstName, string LastName, DateOnly DateOfBirth, string Email, string Phone, string? MiddleName, string? AlternateContactPhone, string? PreferredName);
```
Comment (ytqsl) [inline src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/UserInfoEndpoints.cs:40]: UserInfo does not need these fields

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/UserMapper.cs [hunk 2]
```diff
@@ -12,6 +12,9 @@ public UserMapper()
       .ForCtorParam(nameof(Managers.Registry.Contract.Registrants.UserProfile.DateOfBirth), opts => opts.MapFrom(s => s.DateOfBirth))
       .ForCtorParam(nameof(Managers.Registry.Contract.Registrants.UserProfile.Email), opts => opts.MapFrom(s => s.Email))
       .ForCtorParam(nameof(Managers.Registry.Contract.Registrants.UserProfile.Phone), opts => opts.MapFrom(s => s.Phone))
+      .ForCtorParam(nameof(Managers.Registry.Contract.Registrants.UserProfile.MiddleName), opts => opts.MapFrom(s => s.MiddleName))
```
Comment (ytqsl) [inline src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/UserMapper.cs:15]: Not needed here, essentially you can treat these fields as the Address fields

File: src/ECER.Resources.Accounts/Registrants/RegistrantRepositoryMapper.cs [hunk 3]
```diff
@@ -46,6 +52,7 @@ public RegistrantRepositoryMapper()
         .ForMember(d => d.Address2_PostalCode, opts => opts.MapFrom(s => s.MailingAddress == null ? null : s.MailingAddress.PostalCode))
         .ForMember(d => d.Address2_StateOrProvince, opts => opts.MapFrom(s => s.MailingAddress == null ? null : s.MailingAddress.Province))
         .ForMember(d => d.Address2_Country, opts => opts.MapFrom(s => s.MailingAddress == null ? null : s.MailingAddress.Country))
+
```
Comment (ytqsl) [inline src/ECER.Resources.Accounts/Registrants/RegistrantRepositoryMapper.cs:55]: extra space?

**Review by ytqsl — CHANGES_REQUESTED** [review]

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/UserInfoEndpoints.cs (hunk 1 above)
Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/UserInfoEndpoints.cs]: Removed

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/UserMapper.cs (hunk 2 above)
Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/UserMapper.cs]: @ytqsl, please see my new changes to `UserMapper()`

**Review by ytqsl — APPROVED** [review]

**Pull request merged**